docs/utxo-hd/v0.1/for-developers.md

```diff
@@ -0,0 +1,456 @@
+each block type (i.e. needs the instance `CanStowLedgerTables (LedgerState
+blk)`), and it defines a general flow for **applying a block: a ledger state
+with some values (`ValuesMK`) goes in and a ledger state with the corresponding
+deltas (`DiffMK`) comes out**. This is reflected in the
+`(re)applyBlockLedgerResult` functions:
+
+```haskell
+applyBlockLedgerResult ::
+     HasCallStack
+  => LedgerCfg l
+  -> blk
+  -> Ticked1 l ValuesMK
+  -> Except (LedgerErr l) (LedgerResult l (l DiffMK))
+```
+
+To provide a `LedgerState` with values (`ValuesMK`) to the Ledger, we need to
+follow this sequence of steps:
+
+1. Consult which UTxOs the block will need (`getBlockKeySets`).
+
+2. Read those from the BackingStore (module `ReadsKeySets`).
+
+3. Apply the diffs in the sequence of deltas in memory up to the desired slot (
+   `forwardTableKeySets`).
+
+4. Inject the deltas into the ledger state and call the desired Ledger rule.
+
+This read and forwarding is another of the main sources of speed loss in the
+node. Before UTxO-HD, the values were just in memory already so we didn't need
+to do any further work to bring them in. Now we even have (depending on the
+backing store chosen) to perform I/O to read those.
+
+Here we present a diagram in which the yellow nodes are operations, and the
+`DbChangelog` and `BackingStore` are depicted with colors.
+
+```mermaid
+stateDiagram-v2
+    classDef ledgerdb fill:#556B2F
+    classDef backingstore fill:#8B0000
+    state "block :: blk" as block
+    state "keys :: LedgerTables l KeysMK" as neededEntries
+    block --> getBlockKeySets:::operation
+    dbChangelog:::ledgerdb --> rollback
+    getBlockKeySets --> neededEntries
+    neededEntries --> readKeySets:::operation
+    backingStore:::backingstore --> readKeySets
+    readKeySets --> unforwardedTables
+    state "unforwardedTables :: LedgerTables l ValuesMK" as unforwardedTables
+    block --> rollback:::operation
+    rollback --> truncatedDbChangelog 
+    state "emptyInputState :: l EmptyMK" as emptyInputState
+    truncatedDbChangelog --> current:::operation
+    truncatedDbChangelog --> getDiffs:::operation
+    current --> emptyInputState
+    classDef operation fill:#BDB76B
+    state "seqDiff :: LedgerTables l SeqDiffMK" as seqDiff
+    getDiffs:::dbchangelog --> seqDiff
+    seqDiff --> forwarding:::operation
+    unforwardedTables --> forwarding
+    state "forwardedTables :: LedgerTables l ValuesMK" as forwardedTables
+    forwarding --> forwardedTables
+    emptyInputState --> withLedgerTables:::operation
+    forwardedTables --> withLedgerTables
+    withLedgerTables --> inputState
+    state "inputState :: l ValuesMK" as inputState
+    inputState --> stow:::operation
+    state "stowedState :: l EmptyMK" as stowedState
+    stow --> stowedState
+    block --> apply:::operation
+    stowedState --> apply
+    state "resultingState :: l EmptyMK" as resultingState
+    apply --> resultingState
```

Comment:
I suggest making `apply` a different color, since it's the only node that represents interacting with the ledger.

Or maybe, since `stow`ing is also due to our goal of this first release requiring no changes from ledger, we should include stowing in this color as well?

Perhaps instead of color, you could use a subgraph to draw a box around the stow, apply, and unstow parts.